cbmackintosh/rancid-tomatillos
# Feature/video section component

## What does this PR do?
 - Adds a new VideoSection class component to the component hierarchy
 - Rendered within the MovieDetailsCard, VideoSection retrieves video data for each movie from the server and displays the videos as thumbnails on the card
 - Installs a new dependency: React Player to facilitate cross-site video compatibility. Some movies source their videos from YouTube, others from Vimeo.

## How should it be tested?
 - Fetch this repo and make sure the new dependency is installed `npm install react-player`
 You can find more information about this package at the following links:
https://www.npmjs.com/package/react-player
https://github.com/CookPete/react-player
- Click on any movie poster to see the displayed videos. Check to see if videos are playing as expected. Monitor the console for any error messages. Make sure overall site navigation and functionality is unaffected.
- Run an lighthouse audit to gauge video library's impact on accessibility.

## Next steps
- Noticed some bugs and occasional error messages on the console. I think these can be mostly resolved with improved error handling on VideoSection's network request. See Tenet video number three for an example.
- Styling can obviously be improved, but core functionality for this feature should be good to go.

## Screenshot illustrating new video feature:
![Screen Shot 2021-04-03 at 12 26 29 PM](https://user-images.githubusercontent.com/72054706/113487892-daee4700-9477-11eb-9fbd-0b7a99322220.png)
